[Bug]: Prism runner "received message larger than max" error

### What happened?

While updating [Privacy on Beam](https://github.com/google/differential-privacy/tree/main/privacy-on-beam)'s dependency on Apache Beam Go to version v2.51.0, I am receiving the following error:

```
2023/10/26 15:12:58 stopping worker job-001[go-job-1-1698325959973818582]_go
    distinct_id_test.go:318: DistinctPrivacyID didn't add any Laplace noise: job failed to prepare
        	caused by:
        rpc error: code = ResourceExhausted desc = grpc: received message larger than max (4236507 vs. 4194304)
        error may be due to Apache Beam Go's migration from the direct runner to the prism runner. While the failure(s) should be fixed, you can continue to use the direct runner with this TestMain override: `func TestMain(m *testing.M) { ptest.MainWithDefault(m, "direct") }`
--- FAIL: TestDistinctPrivacyIDAddsNoise (19.35s)
```

on [this](https://github.com/google/differential-privacy/blob/91b4ecebf33e71b8a215b7ea8325fb5e47b12671/privacy-on-beam/pbeam/distinct_id_test.go#L259) test. Curiously, it works fine for all the other tests.

Is this a known issue? Are there any actions I can take to resolve this?

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [X] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/sdks/go/pkg/beam/runners/prism/internal/jobservices/server.go
+++ b/sdks/go/pkg/beam/runners/prism/internal/jobservices/server.go
@@ -17,6 +17,7 @@ package jobservices
 
 import (
 	"fmt"
+	"math"
 	"net"
 	"sync"
 
@@ -60,7 +61,9 @@ func NewServer(port int, execute func(*Job)) *Server {
 		execute: execute,
 	}
 	slog.Info("Serving JobManagement", slog.String("endpoint", s.Endpoint()))
-	var opts []grpc.ServerOption
+	opts := []grpc.ServerOption{
+		grpc.MaxRecvMsgSize(math.MaxInt32),
+	}
 	s.server = grpc.NewServer(opts...)
 	jobpb.RegisterJobServiceServer(s.server, s)
 	jobpb.RegisterArtifactStagingServiceServer(s.server, s)

